File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.java
class Solution {
    public int minIncrementForUnique(int[] nums) {
        Arrays.sort(nums);
        int count = 0;
        for(int i =1; i< nums.length; i++){
            if(nums[i] <= nums[i-1]){
                int extra = nums[i-1] +1 - nums[i];
                count += extra;
                
                nums[i] = nums[i-1] +1;
            }
        }
        return count;
    }
}

File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        max_val = max(nums)
        count = 0
        frq = [0] * (len(nums) + max_val+1)
        
        for val in nums:
            frq[val]+=1
        
        for i in range(len(frq)):
            if(frq[i] <= 1):
                continue
            
            extra = frq[i] -1
            frq[i+1] += extra
            frq[i] = 1
            count += extra
            
        return count 